### PR TITLE
Sync block unlock fix

### LINKED ISSUE #27: error: sync is not blocked
output:
```
13:45:40 [DEBUG] (1) book_safe::sync: sync routes: [34.117.137.206, 34.117.147.117, 142.250.179.180, 142.251.36.20, 142.251.36.52, 142.251.39.116, 151.101.1.195, 151.101.65.195, 216.239.32.21, 216.239.34.21, 216.239.36.21, 216.239.38.21]
13:45:40 [DEBUG] (1) book_safe::sync::route: parsed routes: Ok({34.117.137.206, 192.168.1.0, 0.0.0.0})
13:45:40 [DEBUG] (1) book_safe::sync::route: blocking: 34.117.147.117
13:45:40 [DEBUG] (1) book_safe::sync::route: blocking: 34.117.147.117
13:45:40 [ERROR] Could not block sync
13:45:40 [INFO] starting ui
```

### PR DESCRIPTION
**Pull request by yara-blue**
ready for a code review however still needs testing. Should address all causes of #1

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -175,39 +167,43 @@ fn unlock() -> Result<()> {
...
+        sync::block().wrap_err("Could not block sync")?;
+    }
+    move_docs(&to_lock).wrap_err("Could not move book data")
+}
+
+fn lock(forbidden: Vec<String>, unlock_at: Time, allow_sync: bool) -> Result<()> {
     // if we did not lock the ui before building the file tree the ui could
     // modify the tree while or after we are building it.
     systemd::ui_action("stop").wrap_err("Could not stop gui")?;
-    'lock_or_dont: {
```
Comment (yara-blue) [inline src/main.rs:182]: reverse this, labels are clearer here.

**Comment by yara-blue** [comment]
Do not merge until I have used this branch for a few days. Ready for code review though!

File: src/sync.rs [hunk 2]
```diff
@@ -89,69 +38,124 @@ fn resolve_routes() -> (HashSet<IpAddr>, Vec<ResolveError>) {
...
     (res, err)
 }
 
-fn routes_from_file() -> Result<Vec<IpAddr>> {
-    let text = match fs::read_to_string("routes.txt") {
-        Ok(lines) => lines,
-        Err(e) if e.kind() == io::ErrorKind::NotFound => return Ok(Vec::new()),
-        Err(e) => Err(e).wrap_err("Could not \"routes.txt\"")?,
-    };
+fn sync_routes() -> Result<Vec<IpAddr>> {
```
Comment (evavh) [inline src/sync.rs]: Unclear function name, suggestion: load_and_update_sync_routes

File: src/sync/cache.rs [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+            i += 1;
+        }
+
+        Some(Routes(self.0))
+    }
+}
+
+pub struct Routes(Vec<Entry>);
+
+impl Routes {
```
Comment (evavh) [inline src/sync/cache.rs]: Name doesn't make clear what the difference between Routes and Cached is. Suggestion: UpdatedRoutes?

**Review by evavh — APPROVED** [review]

**Pull request merged**